sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt

```diff
@@ -0,0 +1,135 @@
+package io.sentry.uitest.android.benchmark
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.app.launchActivity
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.action.ViewActions.swipeUp
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.AndroidJUnitRunner
+import io.sentry.ITransaction
+import io.sentry.Sentry
+import io.sentry.SentryOptions
+import io.sentry.android.core.SentryAndroid
+import io.sentry.uitest.android.benchmark.util.BenchmarkOperation
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SentryBenchmarkTest {
```

Comment:
Had a look at it, but:
1) `FrameTimingMetric` and its `frameOverrunMs` require api 31 (or 29 depending on which docs page you open...), and we want to measure against several Android versions
2) Macrobenchmark allows to print the benchmark data to console or to a json, but it's not returned in the code. Comparing two operations and make assertions on them would require a custom gradle plugin, which cannot be used in Saucelabs (not sure if you can use it in firebase), or reading the json results and compare them
Sadly it doesn't have a built-in method to compare two or more operations.
Maybe @indragiek could say something more on it?

Reply:
ok, sounds good. I guess we'll still use macrobenchmark for app startup time, but your approach is useful for when we need to benchmark different options of the sdk.

Reply:
Since `Macrobenchmark` is the latest tooling and has a rich file as output, parsing the JSON file and reading it in the test wouldn't be problematic IMO.
Not telling you to change everything, but I'd go this way if I'd have to choose.

Reply:
API level is a problem indeed, but isn't it about compiling the App and setting the right `compileSdk` and `targetSdk`? or do you mean that tests won't run on older devices?